src/components/VersionChooser/index.spec.tsx

```diff
@@ -299,53 +409,92 @@ describe(__filename, () => {
       version: {
         ...fakeVersion,
         id: headVersionId,
-        // eslint-disable-next-line @typescript-eslint/camelcase
         file: { ...fakeVersionFile, selected_file: selectedFile },
       },
     });
-    _loadVersionsList(store, addonId, fakeVersionsList);
+    _loadVersionsList({ store, addonId });
 
     const history = createFakeHistory();
 
-    const root = render({
+    const { content } = renderForm({
       addonId,
       baseVersionId,
       headVersionId,
       history,
       store,
     });
 
-    const onChange = root
-      .find({ className: styles.headVersionSelect })
-      .prop('onChange');
-    onChange(headVersionId);
+    changeSelectValue(
+      content.find(`.${styles.headVersionSelect}`),
+      headVersionId,
+    );
+
+    submitForm(content);
 
     expect(history.push).toHaveBeenCalledWith(
       `/${lang}/compare/${addonId}/versions/${baseVersionId}...${headVersionId}/?path=${selectedFile}`,
     );
   });
 
-  it('dispatches setCurrentBaseVersionId() when the old version changes', () => {
-    const addonId = nextUniqueId();
-
-    const store = configureStore();
-    _loadVersionsList(store, addonId, fakeVersionsList);
+  it('hides the popover after submitting the form', () => {
+    const { addonId, store } = _loadVersionsList();
     const dispatchSpy = spyOn(store, 'dispatch');
 
-    const selectedVersion = nextUniqueId();
+    const { content } = renderForm({ addonId, store });
 
-    const root = render({ addonId, store });
+    submitForm(content);
 
-    const onChange = root
-      // Retrieve the `VersionSelect` component with this `className`.
-      .find({ className: styles.baseVersionSelect })
-      .prop('onChange');
-    onChange(selectedVersion);
+    expect(dispatchSpy).toHaveBeenCalledWith(popoverActions.hide(POPOVER_ID));
+  });
+
+  it('enables the submit button', () => {
+    const { addonId, store } = _loadVersionsList();
+    const { content } = renderForm({ addonId, store });
+
+    expect(content.find(`.${styles.submitButton}`)).toHaveProp(
+      'disabled',
+      false,
+    );
+  });
+
+  it('disables the submit button when versions are identical', () => {
+    const versionId = nextUniqueId();
+    const { addonId, store } = _loadVersionsList({
+      versions: [
+        { ...fakeListedVersion, id: versionId },
+        { ...fakeListedVersion, id: nextUniqueId() },
+      ],
+    });
 
-    expect(dispatchSpy).toHaveBeenCalledWith(
-      versionsActions.setCurrentBaseVersionId({
-        versionId: selectedVersion,
-      }),
+    const { content, root } = renderForm({
+      addonId,
+      setBaseVersion: false,
+      store,
+    });
+
+    changeSelectValue(content.find(`.${styles.baseVersionSelect}`), versionId);
+
+    changeSelectValue(content.find(`.${styles.headVersionSelect}`), versionId);
+
+    // Re-render the form with updated state.
+    const { content: updatedContent } = simulatePopover(root);
```

Comment:
I guess you need this because you cannot get the UI to select the same version for both base and head. That means the calls to `changeSelectValue` above are somewhat invalid as a user would never be able to do that. Why not just render the form with matching head and base version ids?

Reply:
Ah, yes, I can simplify the test. BTW, you can enter this state in a couple ways. One is to edit the URL to compare the same version to itself. Another is to view an add-on which only has one version (I think).